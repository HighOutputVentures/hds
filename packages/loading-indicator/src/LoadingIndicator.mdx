import { Canvas, Meta, Story, ArgsTable } from '@storybook/addon-docs';

import StorybookHeader from '../../../components/StorybookHeader';

<Meta title="Components/Modal" />

<StorybookHeader title="HDS Modal" command="npm i @highoutput/hds-modal" />

<Canvas>
  <Story id="components-modal--default"></Story>
</Canvas>

<ArgsTable />
