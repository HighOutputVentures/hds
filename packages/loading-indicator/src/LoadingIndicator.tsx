import React, { FC } from 'react';
import { Text, Box, Spinner } from '@chakra-ui/react';

export interface LoadingIndicatorProps {
  variant?: 'solid' | 'twotone';
  size?: 'xs' | 'sm' | 'md' | 'lg' | 'xl';
  showLabel?: boolean;
  labelText?: string;
}

const LoadingIndicator: FC<LoadingIndicatorProps> = ({
  variant = 'solid',
  size = 'md',
  showLabel = true,
  labelText = 'Loading...',
}) => {
  return (
    <Box textAlign="center">
      <Spinner variant={variant} size={size} />
      {showLabel && <Text fontSize="10px">{labelText}</Text>}
    </Box>
  );
};

export default LoadingIndicator;
