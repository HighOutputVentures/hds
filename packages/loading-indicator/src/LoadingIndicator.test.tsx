import { render } from '@testing-library/react';
import LoadingIndicator from './LoadingIndicator';
import * as React from 'react';

describe('LoadingIndicator', () => {
  it('should render', () => {
    const { container } = render(<LoadingIndicator />);

    expect(container).toMatchSnapshot();
  });

  it('should show the label by default', () => {
    const { getByText } = render(<LoadingIndicator />);

    expect(getByText('Loading...')).toBeTruthy();
  });

  it('should update the label', () => {
    const labelText = 'Please wait...';

    const { getByText } = render(<LoadingIndicator labelText={labelText} />);

    expect(getByText(labelText)).toBeTruthy();
  });

  it('should hide the label', () => {
    const { queryByText } = render(<LoadingIndicator showLabel={false} />);

    expect(queryByText('Loading...')).toBeFalsy();
  });
});
