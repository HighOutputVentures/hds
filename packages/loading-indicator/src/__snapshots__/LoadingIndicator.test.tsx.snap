// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingIndicator should render 1`] = `
<div>
  <div
    class="css-w3pvmd"
  >
    <div
      aria-label="animation"
      role="button"
      style="width: 40px; height: 40px; overflow: hidden; margin: 0px auto; outline: none;"
      tabindex="0"
      title=""
    >
      <svg
        height="1080"
        preserveAspectRatio="xMidYMid slice"
        style="width: 100%; height: 100%; transform: translate3d(0,0,0);"
        viewBox="0 0 1080 1080"
        width="1080"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
      >
        <defs>
          <clippath
            id="__lottie_element_2"
          >
            <rect
              height="1080"
              width="1080"
              x="0"
              y="0"
            />
          </clippath>
        </defs>
        <g
          clip-path="url(#__lottie_element_2)"
        >
          <g
            opacity="1"
            style="display: block;"
            transform="matrix(1,0,0,1,520,520)"
          >
            <g
              opacity="1"
              transform="matrix(1,0,0,1,20,20)"
            >
              <path
                d=" M0,346 C0,346 0,346 0,346 C-190.9573974609375,346 -346,190.9573974609375 -346,0 C-346,-190.9573974609375 -190.9573974609375,-346 0,-346"
                fill-opacity="0"
                stroke="rgb(138,104,239)"
                stroke-linecap="round"
                stroke-linejoin="miter"
                stroke-miterlimit="4"
                stroke-opacity="1"
                stroke-width="80"
              />
            </g>
          </g>
        </g>
      </svg>
    </div>
    <p
      class="chakra-text css-9c7r58"
    >
      Loading...
    </p>
  </div>
</div>
`;
